### ISSUE TITLE
New Rule: no-synchronous-wait

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#46]
## Research Evidence

Common LLM mistake: Using `setTimeout` with synchronous expectations or implementing "sleep" incorrectly.

## Description

Detect incorrect async wait patterns like `while` loop busy-waits instead of proper async delays.

### Bad (busy wait)
```ts
function sleep(ms: number) {
  const start = Date.now();
  while (Date.now() - start < ms) {
    // Blocking! Freezes the event loop
  }
}

// Using it
await sleep(1000);
```

### Good
```ts
function sleep(ms: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, ms));
}
```

## Detection Heuristics
- While loops that check elapsed time without yielding
- Synchronous functions named `sleep`, `wait`, `delay`
- Date.now() comparison loops

## Why This Matters for LLMs
- LLMs trained on Python/C++ might suggest blocking sleep
- Blocks Node.js event loop entirely
- Subtle bug that passes type checking

## Priority
Medium - catches dangerous anti-pattern
